src/twisted/internet/test/test_iocp.py

```diff
@@ -64,16 +69,47 @@ def _acceptAddressTest(self, family, localhost):
         client.setblocking(False)
         try:
             client.connect((localhost, port.getsockname()[1]))
-        except error as e:
+        except OSError as e:
             self.assertIn(e.errno, (errno.EINPROGRESS, errno.EWOULDBLOCK))
 
         server = socket(family, SOCK_STREAM)
         self.addCleanup(server.close)
         buff = array("B", b"\0" * 256)
         self.assertEqual(0, _iocp.accept(port.fileno(), server.fileno(), buff, None))
-        server.setsockopt(
-            SOL_SOCKET, SO_UPDATE_ACCEPT_CONTEXT, pack("P", port.fileno())
-        )
+
+        socketError = None
+        for _ in range(5):
+            # Calling setsockopt after _iocp.accept might fail for both IPv4
+            # and IPV6 with [Errno 10057] A request to send or receive ...
+            # This is when ERROR_IO_PENDING is returned and means that the
+            # socket is not yet ready and accept will be handled via the
+            # callback event.
+            # For this test, with the purpose of keeping the test simple,
+            # we don't implement the event callback.
+            # The event callback functionality is tested via the high level
+            # tests for general reactor API.
+            # We retry multiple times to cover.
+            try:
+                server.setsockopt(
+                    SOL_SOCKET, SO_UPDATE_ACCEPT_CONTEXT, pack("P", port.fileno())
+                )
+                # Reset any previous error and stop retrying.
+                socketError = None
+                break
+            except OSError as socketError:
```

Comment:
won't this socketError get deleted?

```
def demo():
    e = None
    try:
        raise Exception
    except Exception as e:
        pass

    return e


demo()
```

```
Traceback (most recent call last):
  File "demo.py", line 11, in <module>
    demo()
  File demo.py", line 8, in demo
    return e
```

Reply:
OMG. Python 3 is crazy :) True

Sorry, I am still working only on Python2.

But why was this error not picked up by the linters.

it looks like a common error to be make when migrating from py2 to py3.

thanks @graingert . Great feedback :)

Reply:
```
for attempts_remaining in range(4, -1, -1):
    try:
        server.setsockopt(SOL_SOCKET, SO_UPDATE_ACCEPT_CONTEXT, pack("P", port.fileno()))
    except OSError as e:
        if attempts_remaining == 0 or e.errno != getattr(errno, "WSAENOTCONN"):
            raise
        time.sleep(0.2)
```

Reply:
will fix this.

More info https://stackoverflow.com/a/44258937/539264 and https://www.python.org/dev/peps/pep-3110/